scripts/update-codegen.sh

```diff
@@ -1,17 +1,44 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 set -o errexit
 set -o nounset
 set -o pipefail
 
 SCRIPT_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
-CODEGEN_PKG=${CODEGEN_PKG:-$(cd "${SCRIPT_ROOT}"; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
-CLIENT_PKG=github.com/aws/aws-app-mesh-controller-for-k8s/pkg/client
-APIS_PKG=github.com/aws/aws-app-mesh-controller-for-k8s/pkg/apis
+ROOT_PKG=github.com/aws/aws-app-mesh-controller-for-k8s
+CLIENT_PKG=${ROOT_PKG}/pkg/client
+APIS_PKG=${ROOT_PKG}/pkg/apis
+
+# Below code is copied from https://github.com/weaveworks/flagger/blob/master/hack/update-codegen.sh
+# Grab code-generator version from go.sum.
+CODEGEN_VERSION=$(grep 'k8s.io/code-generator' go.sum | awk '{print $2}' | head -1)
+CODEGEN_PKG=$(echo `go env GOPATH`"/pkg/mod/k8s.io/code-generator@${CODEGEN_VERSION}")
+
```

Comment:
Can we add a check to make sure the expected module exists at $CODEGEN_PKG and print a descriptive error message with corrective action if it doesn't?  I.e. if the user clones this and immediately tries to run this script, I imagine it would fail, and they could run `go mod download` to fix it.

Reply:
Done

```
➜  aws-app-mesh-controller-for-k8s git:(codegen-fix) ✗ ./scripts/update-codegen.sh
/Users/meduri/go/pkg/mod/k8s.io/code-generator@v0.0.0-20191003035328-700b1226c0bd is missing. Run 'go mod download'.
➜  aws-app-mesh-controller-for-k8s git:(codegen-fix) ✗ go mod download
➜  aws-app-mesh-controller-for-k8s git:(codegen-fix) ✗ ./scripts/update-codegen.sh
>> Using /Users/meduri/go/pkg/mod/k8s.io/code-generator@v0.0.0-20191003035328-700b1226c0bd
>> Temporary output directory /var/folders/51/p6_jj175545g9b6763h13f0cm6rs_6/T/tmp.c3blfmLB
Generating deepcopy funcs
Generating clientset for appmesh:v1beta1 at github.com/aws/aws-app-mesh-controller-for-k8s/pkg/client/clientset
Generating listers for appmesh:v1beta1 at github.com/aws/aws-app-mesh-controller-for-k8s/pkg/client/listers
Generating informers for appmesh:v1beta1 at github.com/aws/aws-app-mesh-controller-for-k8s/pkg/client/informers
>> Removing /var/folders/51/p6_jj175545g9b6763h13f0cm6rs_6/T/tmp.c3blfmLB
```